src/ReTestItems.jl

```diff
@@ -682,6 +682,8 @@ function runtestitem(ti::TestItem, ctx::TestContext; verbose_results::Bool=false
         end
     end
     Test.push_testset(ts)
+    prev = get(task_local_storage(), :__TESTITEM_ACTIVE__, false)
+    task_local_storage()[:__TESTITEM_ACTIVE__] = true
```

Comment:
Can you add a comment maybe explaining a tiny bit about _why_ we might want this hack?